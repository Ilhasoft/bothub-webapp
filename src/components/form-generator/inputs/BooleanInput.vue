<template>
  <b-field
    v-if="compact"
    :label="label"
    horizontal>
    <b-switch
      v-model="value"
      @input="update()" />
    <b-field :message="helpText" />
  </b-field>
  <b-switch
    v-else
    v-model="value"
    @input="update()">{{ verboseValue }}</b-switch>
</template>

<script>

export default {
  props: {
    initialData: {
      type: Boolean,
      default: false,
    },
    compact: {
      type: Boolean,
      default: null,
    },
    label: {
      type: String,
      default: '',
    },
    helpText: {
      type: String,
      default: '',
    },
  },
  data() {
    return {
      value: this.initialData,
    };
  },
  computed: {
    verboseValue() {
      return this.value ? this.$t('webapp.layout.yes') : this.$t('webapp.layout.no');
    },
  },
  mounted() {
    this.update();
  },
  methods: {
    update() {
      this.$emit('input', this.value);
    },
  },
};
</script>
