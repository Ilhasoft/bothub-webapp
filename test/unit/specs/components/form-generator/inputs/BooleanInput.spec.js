import BH from 'bh';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import BooleanInput from '@/components/form-generator/inputs/BooleanInput';

const localVue = createLocalVue();
localVue.use(BH);

describe('BooleanInput.vue', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(BooleanInput);
  });

  test('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
