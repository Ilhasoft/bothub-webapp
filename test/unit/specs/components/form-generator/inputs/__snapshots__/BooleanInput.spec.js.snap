// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BooleanInput.vue renders correctly 1`] = `<b-switch-stub truevalue="true" rounded="true">No</b-switch-stub>`;
